app/models/user.rb

```diff
@@ -16,9 +18,9 @@ class User < ActiveRecord::Base
   attr_accessible *(ACCESSIBLE_ATTRIBUTES + [:admin]), :as => :admin
 
   validates_presence_of :username
-  validates_uniqueness_of :username
+  validates :username, uniqueness: { case_sensitive: false }
```

Comment:
Or just stay away from DB contraints.  The sense of case-folding is different between ruby and the underlying database anyway.